holoviews/plotting/bokeh/callbacks.py

```diff
@@ -195,23 +223,38 @@ def set_customjs(self, handle):
         self_callback = self.js_callback.format(comms_target=self.comm.target,
                                                 timeout=self.timeout,
                                                 debounce=self.debounce)
-        attributes = attributes_js(self.attributes)
-        code = 'var data = {};\n' + attributes + self.code + self_callback
 
         handles = {}
         subplots = list(self.plot.subplots.values())[::-1] if self.plot.subplots else []
         plots = [self.plot] + subplots
         for plot in plots:
             handles.update({k: v for k, v in plot.handles.items()
                             if k in self.handles})
+
+        attributes = attributes_js(self.attributes, handles)
+        code = 'var data = {};\n' + attributes + self.code + self_callback
+
+        # Gather the ids of the plotting handles attached to this callback
+        # This allows checking that a stream is not given the state
+        # of a plotting handle it wasn't attached to
+        stream_handle_ids = defaultdict(list)
+        for stream in self.streams:
+            for h in self.handles:
+                if h in handles:
+                    handle_id = handles[h].ref['id']
+                    stream_handle_ids[stream].append(handle_id)
+
         # Set callback
         if id(handle.callback) in self._callbacks:
             cb = self._callbacks[id(handle.callback)]
             if isinstance(cb, type(self)):
                 cb.streams += self.streams
+                for k, v in stream_handle_ids.items():
+                    cb.stream_handles[k] += v
             else:
                 handle.callback.code += code
         else:
+            self.stream_handles.update(stream_handle_ids)
```

Comment:
Looks like the contents of ``self.stream_handles`` can keep growing between ``set_customjs`` calls with no way to reset/clear it? Is this more state that hangs around when a visualization is removed (e.g a notebook cell is deleted?).